### PR TITLE
feat: use java httpclient instead of apache one

### PR DESCRIPTION
**Pull request by Pawel-608**

### CODE REVIEW

File: common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java [hunk 1]
```diff
@@ -39,6 +37,7 @@
 @RequiredArgsConstructor
 public class AuthService
 {
+    private static final String AUTHORIZATION = "Authorization";
```
Comment (mpichura) [inline common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java:40]: Maybe this constant could be public and reused in `AuthService` from JAMP extension module?

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/util/AuthService.java [hunk 2]
```diff
@@ -37,8 +35,8 @@ public class AuthService
 {
     private final AccessTokenProvider accessTokenProvider;
 
-    public void setAuthHeader(HttpPost request)
+    public String[] getAuthHeaders()
     {
-        request.setHeader(HttpHeaders.AUTHORIZATION, "Bearer " + accessTokenProvider.getAccessToken("hyland-experience"));
+        return new String[]{"Authorization", "Bearer " + accessTokenProvider.getAccessToken("hyland-experience")};
```
Comment (mpichura) [inline prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/util/AuthService.java:40]: See my comment on reusing `AUTHORIZATION` constant.

**Review by mpichura — APPROVED** [review]
LGTM, see my remark on reusing a constant (not crucial).

**Review by tpage-alfresco — APPROVED** [review]
LGTM

**Review by krdabrowski — APPROVED** [review]
LGTM

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/util/AuthService.java (hunk 2 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/util/AuthService.java]: fixed

File: common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java (hunk 1 above)
Comment (Pawel-608) [reply common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java]: fixed

**Pull request merged**